### PR TITLE
Handle KPM_SSO_START cookie.

### PR DESCRIPTION
**Pull request by kaj**
Set or clear it depending on if the user is logged in or not, on the auth callback and on kpm.js.

### CODE REVIEW

File: kpm/kpm-backend/src/widget.js.ts [hunk 1]
```diff
@@ -19,6 +19,13 @@ const distProdProductionPath = "./kpm-frontend/distProd/production";
 export async function widgetJsHandler(req: Request, res: Response) {
   // Check "login_success = false" to avoid infinite loops
   if (req.params.login_success === "false") {
+    res.cookie("KTH_SSO_START", "t", {
+      domain: ".kth.se",
+      maxAge: 4 * 60 * 60,
+      httpOnly: false,
+      secure: true,
+      sameSite: "none",
+    });
```
Comment (jhsware) [inline kpm/kpm-backend/src/widget.js.ts:28]: When is this cookie terminated/cleared during a normal session with login and logout?

**Review by jhsware — CHANGES_REQUESTED** [review]

File: kpm/kpm-backend/src/widget.js.ts (hunk 1 above)
Comment (kaj) [reply kpm/kpm-backend/src/widget.js.ts:28]: On the enxt request to anything social or `kpm.js`.  Social clears it on _every_ request when there is no logged-in user, but I think it's ok to only do it on `kpm.js` here.

Comment (kaj) [reply kpm/kpm-backend/src/widget.js.ts]: Yes, in this place the cookie should be cleared, not set!  😎

File: kpm/kpm-backend/src/auth.ts [hunk 2]
```diff
@@ -177,8 +177,22 @@ auth.post("/callback", async function callbackHandler(req, res, next) {
     if (lang) {
       res.cookie("language", lang, LANG_COOKIE_OPTS);
     }
+    res.cookie("KTH_SSO_START", "t", {
+      domain: ".kth.se",
+      maxAge: 4 * 60 * 60,
+      httpOnly: false,
+      secure: true,
+      sameSite: "none",
+    });
```
[the remark this answers is not in the document]
Comment (kaj) [reply kpm/kpm-backend/src/auth.ts]: Done, with some more clenaup.

File: kpm/kpm-backend/src/auth.ts [hunk 3]
```diff
@@ -189,6 +191,21 @@ auth.post("/callback", async function callbackHandler(req, res, next) {
...
+  sameSite: "none",
+} as const;
+
+export function setSsoCookie(res: Response) {
+  res.cookie("KTH_SSO_START", "t", SSO_COOKIE_OPTIONS);
+}
+export function clearSsoCookie(res: Response) {
+  res.clearCookie("KTH_SSO_START", SSO_COOKIE_OPTIONS);
+}
+
```
Comment (jhsware) [inline kpm/kpm-backend/src/auth.ts:208]: Nice!

**Review by jhsware — APPROVED** [review]
LGTM

**Pull request merged**